### ISSUE TITLE
Think about pairing_data migrations

### ISSUE THREAD

**Issue by Jc2k** [jlusiardi/homekit_python#75]
E.g. going from current stable homekit to add_bluetooth branch fails because:

```
            with open(filename, 'r') as input_fp:
                data = json.load(input_fp)
                for pairing_id in data:
                    if data[pairing_id]['Connection'] == 'IP':
                        self.pairings[pairing_id] = IpPairing(data[pairing_id])
                    else:
                        self.pairings[pairing_id] = BlePairing(data[pairing_id])
```

Existing pairing files won't have a Connection attr. We need to set it to IP for existing records.

**Comment by jlusiardi** [comment]
Valid point! Currently it fails at a missing key in controller.py. I'll try to fix in https://github.com/jlusiardi/homekit_python/tree/fix_75_datamigration

**Comment by jlusiardi** [comment]
Hey @Jc2k,

Do you think the change in #79 is enough for this one?

**Comment by Jc2k** [comment]
Yes I think that’s ok.

In another ticket we talked about automatically refreshing the accessories data if c#/cn changes. I think In both that case and this case the controller should save automatically. But that can be done later.

**Comment by jlusiardi** [comment]
Hey

I would like to try to fix bugs first, missing features second and to optimization then.

Eg remove pairing would be important, with this we could do some automated test cycle to be executed at least on our workstations. Some more automated tests for Travis would also be great.

Of course I will support the efforts to integrate with HASS!

Greets Joachim

**Comment by Jc2k** [comment]
Sure, now I’m getting a better handle on things more tests would be good. I’d like to try and create a mock/fake at the GATT layer so we can test BleSession/BLEPairing better on travis.

But I’ll probably do this on top of my 3 outstanding PRs as it’s likely to be affected by the one that drops staging.GATT quite a bit.

We also need to put reliability on that list! Those RuntimeErrors signify generally the device disconnected before the end of a transaction, usually for a get/put they are retryable, which is something I want to look into. I’ll put that lower down my list than tests. When I see them in HASS it’s usually because the HAP spec says to drop the connection after 30s and HASS polls minutely.

The only optimisation I’ve done so far has really been to mitigate a slow down I introduced, which was by a bug fix (aid getting set to a service Id)

(Actually I’ve probably optimised by accident in my 3rd PR but those were really correctness fixes because of improved GATT wrappers.)

On the HASS front I’m really close to having it listen to connected and disconnected events from the Eve Motion! The event code basically work now, just need to improve how it handles BLE disconnects

I will probably rebase my outstanding branches and look at tests on top of them now.

**Comment by jlusiardi** [comment]
merged #79, so i'll close this one
